src/gpodder/gtkui/main.py

```diff
@@ -1963,125 +1978,57 @@ def treeview_available_show_context_menu(self, treeview, event=None):
             downloaded = all(e.was_downloaded(and_exists=True) for e in episodes)
             downloading = any(e.downloading for e in episodes)
 
-            menu = Gtk.Menu()
+            menu = self.application.builder.get_object('episodes-context')
 
             (open_instead_of_play, can_play, can_download, can_pause, can_cancel, can_delete, can_lock) = self.play_or_download()
 
+            # Play / Stream / Preview / Open
+            menu.remove(0)
             if open_instead_of_play:
-                item = Gtk.ImageMenuItem(_('Open'))
-                item.set_image(Gtk.Image.new_from_icon_name('document-open', Gtk.IconSize.MENU))
+                menu.insert(0, _('Open'), 'win.play')
             else:
                 if downloaded:
-                    item = Gtk.ImageMenuItem(_('Play'))
+                    menu.insert(0, _('Play'), 'win.play')
                 elif downloading:
-                    item = Gtk.ImageMenuItem(_('Preview'))
+                    menu.insert(0, _('Preview'), 'win.play')
                 else:
-                    item = Gtk.ImageMenuItem(_('Stream'))
-                item.set_image(Gtk.Image.new_from_icon_name('media-playback-start', Gtk.IconSize.MENU))
-
-            item.set_sensitive(can_play)
-            item.connect('activate', self.on_playback_selected_episodes)
-            menu.append(item)
-
-            if can_download:
-                item = Gtk.ImageMenuItem(_('Download'))
-                item.set_image(Gtk.Image.new_from_icon_name('document-save', Gtk.IconSize.MENU))
-                item.set_action_name('win.download')
-                menu.append(item)
+                    menu.insert(0, _('Stream'), 'win.play')
+
+            # Download / Pause / Cancel
```

Comment:
The Download and Pause items can be shared, but Cancel needs its own item. This is because Download resumes a paused download and Cancel cancels the download if paused or downloading.